cordova/cordova-feature-android-tests/feature/comm.py

```diff
@@ -79,42 +79,69 @@ def setUp():
         VERSION = "3.6"
     f_version.close()
 
-def create(cmd, appname, self):
-    os.chdir(pack_tool)
-    if os.path.exists(os.path.join(pack_tool, appname)):
+def create(appname, pkgname, mode, self):
+    os.chdir(tool_path)
+    if os.path.exists(os.path.join(tool_path, appname)):
         print "Existing %s project, try to clean up..." % appname
-        do_remove(glob.glob(os.path.join(pack_tool, appname)))
+        do_remove(glob.glob(os.path.join(tool_path, appname)))
     print "Create project %s ----------------> START" % appname
-    print cmd
+    if VERSION == "4.0":
+        cmd = "cordova create %s %s %s" % (appname, pkgname, appname)
+    else:
+        if mode == "shared":
+            cmd = "cordova/bin/create %s %s %s --xwalk-shared-library" % (appname, pkgname, appname)
+        else:
+            cmd = "cordova/bin/create %s %s %s" % (appname, pkgname, appname)
     createstatus = commands.getstatusoutput(cmd)
     self.assertEquals(0, createstatus[0])
     print "\nGenerate project %s ----------------> OK\n" % appname
     result = commands.getstatusoutput("ls")
     self.assertIn(appname, result[1])
     if VERSION == "4.0":
+        project_root = os.path.join(tool_path, appname)
+        os.chdir(project_root)
+        replaceUserString(project_root, 'config.xml', '<widget', '<widget android-activityName="%s"' % appname)
+        replaceUserString(project_root, 'config.xml', '</widget>', '    <allow-navigation href="*" />\n</widget>')
+
+        print "Add android platforms to this project --------------> START"
+        cordova_platform_cmd = "cordova platform add android"
+        platformstatus = commands.getstatusoutput(cordova_platform_cmd)
+        self.assertEquals(0, platformstatus[0])
+
         print "Install Crosswalk WebView Plugin --------------> START"
-        os.chdir(os.path.join(pack_tool, appname))
-        plugin_install_cmd = "plugman install --platform android --project ./ --plugin %s" % plugin_tool
+        plugin_install_cmd = "cordova plugin add %s" % plugin_tool
         pluginstatus = commands.getstatusoutput(plugin_install_cmd)
         self.assertEquals(0, pluginstatus[0])
 
-def build(cmd, appname, self):
-    os.chdir(os.path.join(pack_tool, appname))
+def build(appname, self):
+    os.chdir(os.path.join(tool_path, appname))
     print "Build project %s ----------------> START" % appname
+    if VERSION == "4.0":
+        cmd = "cordova build android"
+        if checkContains(appname, "REMOTEDEBUGGING"):
```

Comment:
Please make this as a common function, without hard code. You can make the debug as an argument.